controllers/home_test.go

```diff
@@ -6,18 +6,46 @@ import (
 	"strings"
 	"testing"
 
+	"github.com/gutakk/go-google-scraper/db"
+	"github.com/gutakk/go-google-scraper/models"
 	"github.com/gutakk/go-google-scraper/tests"
 	"gopkg.in/go-playground/assert.v1"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
 )
 
-func TestDisplayHome(t *testing.T) {
+func TestDisplayHomeWithoutUserSession(t *testing.T) {
 	engine := tests.GetRouter(true)
 	new(HomeController).applyRoutes(engine)
 
 	w := tests.PerformRequest(engine, "GET", "/", nil, nil)
 	p, err := ioutil.ReadAll(w.Body)
-	pageOK := err == nil && strings.Index(string(p), "<title>Home</title>") > 0
+	isHomePage := err == nil && strings.Index(string(p), "<title>Home</title>") > 0
 
 	assert.Equal(t, http.StatusOK, w.Code)
-	assert.Equal(t, true, pageOK)
+	assert.Equal(t, true, isHomePage)
+}
+
+func TestDisplayHomeWithUserSession(t *testing.T) {
+	testDB, _ := gorm.Open(postgres.Open(tests.ConstructTestDsn()), &gorm.Config{})
+	db.GetDB = func() *gorm.DB {
+		return testDB
+	}
+
+	_ = db.GetDB().AutoMigrate(&models.User{})
+
+	engine := tests.GetRouter(true)
+	new(HomeController).applyRoutes(engine)
+
+	// Cookie from login API Set-Cookie header
+	headers := http.Header{}
+	cookie := "mysession=MTYwNjI3ODk0NHxEdi1CQkFFQ180SUFBUkFCRUFBQUlmLUNBQUVHYzNSeWFXNW5EQWtBQjNWelpYSmZhV1FFZFdsdWRBWUVBUDRCR0E9PXxa_dKXde8j6m4z_kPgaiPYuDGHj79HxhCMNw3zIoeM6g=="
```

Comment:
What is the validity of this cookie? Can this expire and break tests later?

Reply:
Nice catch. I'm not really sure about this as the whole cookie look like this
```
mysession=MTYwNjQ0OTkxN3xEdi1CQkFFQ180SUFBUkFCRUFBQUlmLUNBQUVHYzNSeWFXNW5EQWtBQjNWelpYSmZhV1FFZFdsdWRBWUVBUDRFS2c9PXxsvz7BbGtpuEyUB7G0IUTjxgBAEkBNd10pBS1NGZEtOA==; Path=/; Expires=Sun, 27 Dec 2020 04:05:17 GMT; Max-Age=2592000```

Not sure if the cookie is valid when I don't set an `Expires`.
Do you have any idea?

Reply:
Likely since the `Expires` header is not set, this should work all the time. By the way, even in the application, the expiration is not defined so maybe adding the Options for MaxAge would be a good addition:

```go
func Set(c *gin.Context, key string, value interface{}) {
    session := sessions.Default(c)
    session.Options(sessions.Options{
        MaxAge: 3600*24,  // 24 hours
    })
	session.Set(key, value)
	_ = session.Save()
}
```

Also in terms of key naming `mysession` could be changed to the application name 💭

Reply:
Update session name in 4b067f4557366e5a1d29960505c4f068545b3c87
Set max age for session in bf2c595ab42740c4fc30dc3cb09e65f5ce451b11